modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptDocsGenerator.cs

```diff
@@ -0,0 +1,268 @@
+
+                    AppendPartialContainingTypeDeclarations(containingType.ContainingType);
+
+                    source.Append("partial ");
+                    source.Append(containingType.GetDeclarationKeyword());
+                    source.Append(" ");
+                    source.Append(containingType.NameWithTypeParameters());
+                    source.Append("\n{\n");
+                }
+            }
+
+            source.Append("partial class ");
+            source.Append(symbol.NameWithTypeParameters());
+            source.Append("\n{\n");
+
+            source.Append("#pragma warning disable CS0109 // Disable warning about redundant 'new' keyword\n");
+            source.Append("#if TOOLS\n");
+
+            source.Append("    [global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]\n");
+            source.Append("    internal new static global::Godot.Collections.Dictionary GetGodotClassDocs()\n    {\n");
+            source.Append("        var docs = new global::Godot.Collections.Dictionary();\n");
+            source.Append("        docs.Add(\"name\",\"");
+            if (symbol.GetAttributes().Any(a => a.AttributeClass?.IsGodotGlobalClassAttribute() ?? false))
+            {
+                source.Append(symbol.Name);
+            }
+            else
+            {
+                source.Append("\\\"" + ScriptPathAttributeGenerator.RelativeToDir(symbol.DeclaringSyntaxReferences.First().SyntaxTree.FilePath, godotProjectDir) + "\\\"");
+            }
+            source.Append("\");\n");
+            source.Append("        docs.Add(\"description\",\"");
+            source.Append(classDescription);
+            source.Append("\");\n\n");
+
+            if (exportedProperties.Length + exportedFields.Length > 0)
+            {
+                source.Append("        var propertyDocs = new global::Godot.Collections.Array();\n");
+                source.Append(docExportedString);
+                source.Append("        docs.Add(\"properties\", propertyDocs);\n\n");
+            }
+
+            if (signalDelegateSymbols.Length > 0)
+            {
+                source.Append("        var signalDocs  = new global::Godot.Collections.Array();\n");
+                source.Append(docSignalString);
+                source.Append("        docs.Add(\"signals\", signalDocs);\n\n");
+            }
+
+            source.Append("        return docs;\n    }\n\n");
+
+            source.Append("#endif // TOOLS\n");
+
+            source.Append("#pragma warning restore CS0109\n");
+
+            source.Append("}\n"); // partial class
+
+            if (isInnerClass)
+            {
+                var containingType = symbol.ContainingType;
+
+                while (containingType != null)
+                {
+                    source.Append("}\n"); // outer class
+
+                    containingType = containingType.ContainingType;
+                }
+            }
```

Comment:
I'm not sure if `DocData` supports nested classes. If it doesn't, we can bail out earlier.